src/step-function.ts

```diff
@@ -1579,9 +1579,9 @@ function getStepFunctionArgs<
     | [func: StepFunctionClosure<Payload, Out>]
 ) {
   const props =
-    isFunctionLike(args[0]) || isErr(args[0])
+    isFunctionLike(args[0]) || isErr(args[0]) || typeof args[0] === "function"
       ? {}
-      : (args[1] as StepFunctionProps);
+      : args[0];
```

Comment:
What case is this catching? When would this be a function?

Reply:
the actual change was changing the index from `1` to `0`, the function check narrows the types it comes in as without using `as`, it will actually be expressions, but the type on it is a function.